src/fsharp/MethodOverrides.fs

```diff
@@ -460,29 +460,34 @@ module DispatchSlotChecking =
                                          else -1
                                      )
-
-                                 // TODO: Add check for most specific implementation when an explicit implementation is given.
-                                 let mutable hasMostSpecificImplementation = true
-                                 for (ty1, RequiredSlot (_, isOptional1)) in sortedSlots do
-                                    if hasMostSpecificImplementation && isOptional1 then
-                                        for (ty2, RequiredSlot (_, isOptional2)) in sortedSlots do
-                                            if isOptional2 &&
-                                               not (TypeFeasiblySubsumesType 0 g amap reqdTyRange ty1 CanCoerce ty2) && 
-                                               not (TypeFeasiblySubsumesType 0 g amap reqdTyRange ty2 CanCoerce ty1) then
-                                                hasMostSpecificImplementation <- false
-                                                // TODO: Move this to SR
-                                                errorR(Error((5000, sprintf "Interface member %A does not have a most specific implementation." (NicePrint.stringOfMethInfo amap reqdTyRange denv reqdSlot)), reqdTyRange))
-
-                                 if hasMostSpecificImplementation then                                 
-                                     match sortedSlots with
-                                     | [] ->
-                                         yield RequiredSlot(reqdSlot, not reqdSlot.IsAbstract)
-                                     | (_, bestSlot) :: _ ->                                      
-                                         yield bestSlot
-                                 else
-                                     // We make this optional because we already error'ed due to no most specific override.
-                                     // Try not to show more errors.
-                                     yield RequiredSlot(reqdSlot, true)
+                                
+                                 match sortedSlots with
+                                 | [] ->
+                                    yield RequiredSlot(reqdSlot, not reqdSlot.IsAbstract)
+                                 | (ty1, (RequiredSlot (_, isOptional1) as bestSlot)) :: sortedSlotsTail ->
+
+                                    // TODO: Add check for most specific implementation when an explicit implementation is given.
+                                    let mutable hasMostSpecificImplementation = true
+                                    let mutable isOptional1 = isOptional1
+                                    for (ty2, RequiredSlot (_, isOptional2)) in sortedSlotsTail do
+
+                                        let isNotPartOfHierarchy =
+                                            not (TypeFeasiblySubsumesType 0 g amap reqdTyRange ty1 CanCoerce ty2) && 
```

Comment:
What is the meaning of `0` here? Should that be factored into a value somewhere?

Reply:
I will need to look. But all calls to this took 0 in other checks, I just copy pasted it.